streams/src/main/java/org/apache/kafka/streams/processor/internals/StateManagerUtil.java

```diff
@@ -115,7 +115,7 @@ static void closeStateManager(final Logger log,
                                   final StateDirectory stateDirectory,
                                   final TaskType taskType) {
         // if EOS is enabled, wipe out the whole state store for unclean close since it is now invalid
-        final boolean wipeStateStore = !closeClean && eosEnabled;
+        final boolean wipeStateStore = !closeClean && eosEnabled && stateMgr.transactional();
```

Comment:
Shouldn't this be `&& !stateMgr.transactional()`?

Reply:
Excellent catch. Thank you!